pkg/config/config.go

```diff
@@ -347,6 +355,160 @@ func (m *Metric) validateMetric(metricIdx int, parent string, discovery *JobLeve
+				Delay:                  metric.Delay,
+				NilToZero:              metric.NilToZero,
+				AddCloudwatchTimestamp: metric.AddCloudwatchTimestamp,
+			})
+		}
+
+		jobsCfg.DiscoveryJobs = append(jobsCfg.DiscoveryJobs, job)
+	}
+
+	for _, staticJob := range c.Static {
+		job := model.StaticJob{}
+		job.Name = staticJob.Name
+		job.Namespace = staticJob.Namespace
+		job.Regions = staticJob.Regions
+
+		for _, role := range staticJob.Roles {
+			job.Roles = append(job.Roles, model.Role{
+				RoleArn:    role.RoleArn,
+				ExternalID: role.ExternalID,
+			})
+		}
+
+		for _, customTag := range staticJob.CustomTags {
+			job.CustomTags = append(job.CustomTags, model.Tag{
+				Key:   customTag.Key,
+				Value: customTag.Value,
+			})
+		}
+
+		for _, dimension := range staticJob.Dimensions {
+			job.Dimensions = append(job.Dimensions, model.Dimension{
+				Name:  dimension.Name,
+				Value: dimension.Value,
+			})
+		}
+
+		for _, metric := range staticJob.Metrics {
+			job.Metrics = append(job.Metrics, &model.MetricConfig{
+				Name:                   metric.Name,
+				Statistics:             metric.Statistics,
+				Period:                 metric.Period,
+				Length:                 metric.Length,
+				Delay:                  metric.Delay,
+				NilToZero:              metric.NilToZero,
+				AddCloudwatchTimestamp: metric.AddCloudwatchTimestamp,
+			})
+		}
+
+		jobsCfg.StaticJobs = append(jobsCfg.StaticJobs, job)
+	}
+
+	for _, customNamespaceJob := range c.CustomNamespace {
+		job := model.CustomNamespaceJob{}
+
+		job.Regions = customNamespaceJob.Regions
+		job.Name = customNamespaceJob.Name
+		job.Namespace = customNamespaceJob.Namespace
+		job.DimensionNameRequirements = customNamespaceJob.DimensionNameRequirements
+		job.RoundingPeriod = customNamespaceJob.RoundingPeriod
+		job.RecentlyActiveOnly = customNamespaceJob.RecentlyActiveOnly
+		job.Statistics = customNamespaceJob.Statistics
+		job.Period = customNamespaceJob.Period
+		job.Length = customNamespaceJob.Length
+		job.Delay = customNamespaceJob.Delay
+		job.NilToZero = customNamespaceJob.NilToZero
+		job.AddCloudwatchTimestamp = customNamespaceJob.AddCloudwatchTimestamp
+
+		for _, role := range customNamespaceJob.Roles {
+			job.Roles = append(job.Roles, model.Role{
+				RoleArn:    role.RoleArn,
+				ExternalID: role.ExternalID,
+			})
+		}
+
+		for _, customTag := range customNamespaceJob.CustomTags {
+			job.CustomTags = append(job.CustomTags, model.Tag{
+				Key:   customTag.Key,
+				Value: customTag.Value,
+			})
+		}
+
+		for _, metric := range customNamespaceJob.Metrics {
+			job.Metrics = append(job.Metrics, &model.MetricConfig{
+				Name:                   metric.Name,
+				Statistics:             metric.Statistics,
+				Period:                 metric.Period,
+				Length:                 metric.Length,
+				Delay:                  metric.Delay,
+				NilToZero:              metric.NilToZero,
+				AddCloudwatchTimestamp: metric.AddCloudwatchTimestamp,
+			})
+		}
```

Comment:
If I'm reading this right it looks like these three mappings show up for each job type. Thoughts on a making a `toModelRole`, `toModelTag`, and `toModelMetricConfig`?

Reply:
Good call, done!